src/oscar/apps/customer/utils.py

```diff
@@ -132,9 +132,12 @@ def get_password_reset_url(user, token_generator=default_token_generator):
     """
     Generate a password-reset URL for a given user
     """
+    uidb64 = urlsafe_base64_encode(force_bytes(user.id))
+    if isinstance(uidb64, bytes):
```

Comment:
We will almost certainly forget to remove this check when we drop support for Django 2.1 and below. I think it is better to wrap `urlsafe_base64_encode` in a helper method in `oscar.core.compat`, so we know it can be removed later without leaving dead code around.

Reply:
Ok done.